From b8b5bcab59fc3e8ce879dbe821716481dc380bf1 Mon Sep 17 00:00:00 2001
Message-ID: <b8b5bcab59fc3e8ce879dbe821716481dc380bf1.1713654571.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1713654571.git.sharlatanus@gmail.com>
References: <cover.1713654571.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 20 Apr 2024 21:53:00 +0100
Subject: [PATCH 42/47] gnu: python-asdf: Adjust inputs.

python-asdf requires python-semantic-version during runtime which may be
reproduced by:

   guix shell python-asdf python -- python3 -c 'import asdf'
   <...>
   ModuleNotFoundError: No module named 'semantic_version'

* gnu/packages/astronomy.scm (python-asdf): Adjust inputs.
[native-inputs]: Remove python-semantic-version.
[propagated-inputs]: Add python-semantic-version.

(python-asdf-astropy, python-asdf-compression, python-asdf-coordinates-schemas,
python-asdf-wcs-schemas, python-asdf-zarr, python-astroalignn, python-ccdproc,
python-crds, python-ginga, python-gwcs, python-photutils, python-rad,
python-reproject, python-roman-datamodels, python-spectral-cube,
python-specutils, python-stcal, python-stdatamodels, python-stpipe,
python-stsynphot, python-sunpy, python-synphot, python-tweakwcs): Remove
python-semantic-version from native or propagated inputs.

Change-Id: I96bf57f77c327f4ff208e0b173e84bcdd4c287da
Reported-by: Iliya Tikhonenko <tikhonenko@mpe.mpg.de>
---
 gnu/packages/astronomy.scm | 41 +++++++++-----------------------------
 1 file changed, 9 insertions(+), 32 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 58bf607059..5cb62466cb 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -522,8 +522,7 @@ (define-public python-aplpy
            python-shapely))
     (native-inputs
      (list python-pytest-astropy
-           python-pytest-mpl
-           python-semantic-version))
+           python-pytest-mpl))
     (home-page "http://aplpy.github.io")
     (synopsis "Astronomical Plotting Library in Python")
     (description
@@ -753,7 +752,6 @@ (define-public python-ginga
            python-docutils
            python-pytest-astropy
            python-pytest-astropy-header
-           python-semantic-version
            python-tornado))
     (home-page "https://ejeschke.github.io/ginga/")
     (synopsis "Scientific image viewer and toolkit for FITS files")
@@ -2014,8 +2012,7 @@ (define-public python-ccdproc
     (build-system pyproject-build-system)
     (native-inputs
      (list python-memory-profiler
-           python-pytest-astropy
-           python-semantic-version))
+           python-pytest-astropy))
     (propagated-inputs
      (list python-astropy
            python-astroscrappy
@@ -2138,7 +2135,6 @@ (define-public python-crds
                          python-nose
                          python-pylint
                          python-pytest
-                         python-semantic-version
                          python-setuptools-scm))
     (home-page "https://hst-crds.stsci.edu")
     (synopsis "Calibration Reference Data System for HST and JWST")
@@ -2481,7 +2477,6 @@ (define-public python-photutils
            python-extension-helpers
            python-pytest-astropy
            python-pytest-xdist
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://github.com/astropy/photutils")
     (synopsis "Source detection and photometry")
@@ -2792,7 +2787,6 @@ (define-public python-reproject
            python-pytest-astropy
            python-pytest-xdist
            ;; python-sunpy ; circular dependencies, test optional
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://reproject.readthedocs.io")
     (synopsis "Astronomical image reprojection in Python")
@@ -2909,7 +2903,6 @@ (define-public python-sunpy
            python-reproject
            python-scikit-image
            python-scipy
-           python-semantic-version
            ;; python-spiceypy ; Not packed yet in Guix, long jorney.
            python-sqlalchemy
            python-tqdm
@@ -3069,7 +3062,6 @@ (define-public python-spectral-cube
            python-pytest-astropy
            python-pytest-xdist
            python-regions
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://spectral-cube.readthedocs.io/en/latest/")
     (synopsis "Library for reading and analyzing astrophysical spectral data cubes")
@@ -3124,7 +3116,6 @@ (define-public python-specutils
     (native-inputs
      (list python-matplotlib
            python-pytest-astropy
-           python-semantic-version
            python-setuptools-scm
            python-spectral-cube))
     (home-page "https://specutils.readthedocs.io/")
@@ -3322,7 +3313,6 @@ (define-public python-stcal
            python-pytest
            python-pytest-cov
            python-pytest-doctestplus
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/stcal")
     (synopsis "STScI tools and algorithms used in calibration pipelines")
@@ -3364,7 +3354,6 @@ (define-public python-stdatamodels
            python-pytest-doctestplus
            python-pytest-xdist
            python-scipy
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/stdatamodels")
     (synopsis
@@ -3390,8 +3379,8 @@ (define-public python-stpipe
       ;; See https://github.com/spacetelescope/stpipe/issues/114
       #:test-flags #~(list "-k" "not test_roman_datamodel")))
     (build-system pyproject-build-system)
-    (propagated-inputs (list python-asdf python-astropy python-crds
-                             python-semantic-version python-stdatamodels))
+    (propagated-inputs
+     (list python-asdf python-astropy python-crds python-stdatamodels))
     (native-inputs (list python-pytest python-pytest-doctestplus
                          python-pytest-openfiles python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/stpipe")
@@ -3429,7 +3418,6 @@ (define-public python-stsynphot
     (native-inputs (list python-pytest
                          python-pytest-astropy
                          python-pytest-astropy-header
-                         python-semantic-version
                          python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/stsynphot_refactor")
     (synopsis "Synthetic photometry using Astropy for HST and JWST")
@@ -4423,7 +4411,6 @@ (define-public python-synphot
     (native-inputs
      (list python-pytest
            python-pytest-astropy
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/synphot_refactor")
     (synopsis "Synthetic photometry using Astropy")
@@ -4457,7 +4444,6 @@ (define-public python-tweakwcs
            python-pytest
            python-pytest-cov
            python-scipy
-           python-semantic-version
            python-setuptools-scm))
     (home-page "https://tweakwcs.readthedocs.io/en/latest/")
     (synopsis
@@ -4503,7 +4489,6 @@ (define-public python-asdf
            python-pytest-doctestplus
            python-pytest-remotedata
            python-pytest-xdist
-           python-semantic-version
            python-setuptools-scm))
     (propagated-inputs
      (list python-asdf-standard
@@ -4513,7 +4498,8 @@ (define-public python-asdf
            python-jmespath
            python-lz4
            python-numpy
-           python-pyyaml))
+           python-pyyaml
+           python-semantic-version))
     (home-page "https://github.com/asdf-format/asdf")
     (synopsis "Python tools to handle ASDF files")
     (description
@@ -4549,7 +4535,6 @@ (define-public python-asdf-compression
       (native-inputs
        (list python-numpy
              python-pytest
-             python-semantic-version
              python-setuptools-scm))
       (propagated-inputs
        (list python-asdf
@@ -4648,7 +4633,6 @@ (define python-asdf-coordinates-schemas
     (native-inputs
      (list python-pytest
            python-pytest-xdist
-           python-semantic-version
            python-setuptools-scm))
     (propagated-inputs
      (list python-asdf))
@@ -4767,7 +4751,7 @@ (define-public python-asdf-zarr
     (propagated-inputs
      (list python-asdf python-fsspec python-zarr))
     (native-inputs
-     (list python-pytest python-semantic-version python-setuptools-scm))
+     (list python-pytest python-setuptools-scm))
     (home-page "https://github.com/asdf-format/asdf-zarr")
     (synopsis "Asdf extension to support Zarr arrays")
     (description
@@ -4802,7 +4786,6 @@ (define-public python-asdf-astropy
            python-pandas
            python-pytest-astropy
            python-scipy
-           python-semantic-version
            python-setuptools-scm))
     (propagated-inputs
      (list python-asdf
@@ -4836,7 +4819,6 @@ (define python-asdf-wcs-schemas
      (list python-asdf
            python-pytest
            python-pytest-openfiles
-           python-semantic-version
            python-setuptools-scm))
     (propagated-inputs
      (list python-asdf-coordinates-schemas
@@ -4867,7 +4849,6 @@ (define-public python-gwcs
            python-pytest
            python-pytest-doctestplus
            python-pyyaml
-           python-semantic-version
            python-setuptools-scm))
     (propagated-inputs
      (list python-asdf
@@ -4905,9 +4886,7 @@ (define-public python-rad
       ;; python-rad -> python-roman-datamodels -> python-crds -> python-rad
       #:test-flags #~(list "--ignore=tests/test_schemas.py")))
     (native-inputs
-     (list python-pytest
-           python-semantic-version
-           python-setuptools-scm))
+     (list python-pytest python-setuptools-scm))
     (propagated-inputs (list python-asdf python-asdf-astropy))
     (home-page "https://github.com/spacetelescope/rad")
     (synopsis "Roman Attribute Dictionary")
@@ -5021,7 +5000,6 @@ (define-public python-roman-datamodels
                          python-pytest-doctestplus
                          python-pytest-env
                          python-pytest-xdist
-                         python-semantic-version
                          python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/roman_datamodels")
     (synopsis "Roman Datamodels Support")
@@ -5055,8 +5033,7 @@ (define-public python-astroalign
      (list python-astropy
            python-ccdproc
            python-pillow
-           python-pytest
-           python-semantic-version))
+           python-pytest))
     (propagated-inputs
      (list python-bottleneck
            python-numpy
-- 
2.41.0

